### DISCUSSION TITLE
How do we import LDAP groups using the `artifactory_ldap_group_setting` resource

### DISCUSSION THREAD

**Discussion by EkhatorOwen** [jfrog/terraform-provider-artifactory#767] [General] [answered]
the LDAP group resource allows for creating the group, however, I still have to click the `import` button in the UI to get the corresponding LDAP group imported. is there a way to do this via terraform so i don't have to click into the UI?

**Comment by alexhung** [comment]
@EkhatorOwen Sorry for the late reply. Let me take a look and get back to you.

**Reply by alexhung** [reply, marked as answer]
As far as I can tell, there is currently no API support to synchronize and import LDAP groups. Please open a feature request ticket with JFrog customer support. Once the API is released, we can add the support to the provider.

**Reply by alexhung** [reply]
The [REST API](https://jfrog.com/help/r/jfrog-rest-apis/update-ldap-group-settings) to import LDAP group settings was added back in Feb 2024. I didn't notice it until now. This has been added to our plan to implement.

**Comment by frednntelia** [comment]
> the LDAP group resource allows for creating the group, however, I still have to click the `import` button in the UI to get the corresponding LDAP group imported. is there a way to do this via terraform so i don't have to click into the UI?

It's possible to emulate the ajax call that the UI makes to the backend, as long as you can obtain a refresh token to pass with it:

```bash
curl 'https://YOUR_JFROG_HOSTNAME/ui/api/v1/access/api/ui/ldap/groups/import' \
  -H 'Content-Type: application/json' \
  -H 'Cookie: ACCESSTOKEN='${ACCESS_TOKEN}'; REFRESHTOKEN='${REFRESH_TOKEN} \
  -H 'X-Requested-With: XMLHttpRequest' \
  --data-raw '{THE PAYLOAD THAT YOU SEE WHEN YOU INSPECT THE UIs IMPORT-GROUP REQUEST IN YOUR BROWSER}'
  ```
